File: adapters/php/src/Symfony/FlexDocController.php
<?php

declare(strict_types=1);

namespace Prauga\FlexDoc\Symfony;

use Prauga\FlexDoc\FlexDocHost;
use Prauga\FlexDoc\FlexDocResponse;
use Symfony\Component\HttpFoundation\Response;

final class FlexDocController
{
    public function __construct(private readonly FlexDocHost $host) {}

    public function documentation(): Response { return $this->response($this->host->documentation()); }
    public function rendererJavaScript(): Response { return $this->response($this->host->rendererJavaScript()); }
    public function rendererCss(): Response { return $this->response($this->host->rendererCss()); }

    private function response(FlexDocResponse $response): Response
    {
        return new Response($response->body, $response->status, $response->headers());
    }
}

File: adapters/java-spring/src/main/java/com/prauga/flexdoc/spring/FlexDocController.java
package com.prauga.flexdoc.spring;

import com.prauga.flexdoc.jvm.FlexDocHost;
import com.prauga.flexdoc.jvm.FlexDocHttpResponse;
import org.springframework.http.HttpHeaders;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

/** Spring MVC transport wrapper around the framework-neutral FlexDoc JVM host. */
@RestController
public class FlexDocController {
  private final FlexDocProperties properties;
  private final FlexDocHost host;

  public FlexDocController(FlexDocProperties properties, FlexDocHost host) {
    this.properties = properties;
    this.host = host;
  }

  @GetMapping(value = "${flexdoc.path:/docs}", produces = MediaType.TEXT_HTML_VALUE)
  public ResponseEntity<byte[]> documentation() throws Exception { return response(host.documentation()); }

  @GetMapping(value = "${flexdoc.path:/docs}/__flexdoc/renderer.js", produces = "application/javascript")
  public ResponseEntity<byte[]> rendererJavaScript() { return response(host.rendererJavaScript()); }

  @GetMapping(value = "${flexdoc.path:/docs}/__flexdoc/renderer.css", produces = "text/css")
  public ResponseEntity<byte[]> rendererCss() { return response(host.rendererCss()); }

  private ResponseEntity<byte[]> response(FlexDocHttpResponse response) {
    return ResponseEntity.status(response.status())
        .header(HttpHeaders.CACHE_CONTROL, response.cacheControl())
        .contentType(MediaType.parseMediaType(response.contentType()))
        .body(response.body());
  }
}
